guides/getting-started/using-microsoft-dev-tunnels.md

```diff
@@ -0,0 +1,90 @@
+
+## Microsoft dev tunnels
+
+Microsoft dev tunnels are available for [Visual Studio Code](https://code.visualstudio.com/docs/editor/port-forwarding) and [Visual Studio](https://learn.microsoft.com/en-us/aspnet/core/test/dev-tunnels?view=aspnetcore-8.0)
+
+Explaining these tools in detail is outside the scope of this guide. Please refer to the following links to
+- [Create a dev tunnel in Visual Studio Code](https://code.visualstudio.com/docs/editor/port-forwarding)
+- [Create a dev tunnel in Visual Studio 2022](https://learn.microsoft.com/en-us/aspnet/core/test/dev-tunnels?view=aspnetcore-8.0)
+
+### Configure the IssuerUrl
+
+In order to use this tunnel now, you need to set the `IssuerUrl` in the client project (e.g. `MauiProgram.cs`)
+```csharp
+    // Register the OpenIddict client components.
+    .AddClient(options =>
+    {
+        // ...
+
+        // Add a client registration matching the client application definition in the server project.
+        options.AddRegistration(new OpenIddictClientRegistration
+        {
+            Issuer = new Uri("https://localhost:44395/", UriKind.Absolute), // [!code --]
+            Issuer = new Uri("https://vsr1d2gg-44359.euw.devtunnels.ms/", UriKind.Absolute), // [!code ++]
+            ProviderName = "Local",
+
+            ClientId = "maui",
+
+            // This sample uses protocol activations with a custom URI scheme to handle callbacks.
+            //
+            // For more information on how to construct private-use URI schemes,
+            // read https://www.rfc-editor.org/rfc/rfc8252#section-7.1 and
+            // https://www.rfc-editor.org/rfc/rfc7595#section-3.8.
+            PostLogoutRedirectUri = new Uri("com.openiddict.sandbox.maui.client:/callback/logout/local", UriKind.Absolute),
+            RedirectUri = new Uri("com.openiddict.sandbox.maui.client:/callback/login/local", UriKind.Absolute),
+
+            Scopes = { Scopes.Email, Scopes.Profile, Scopes.OfflineAccess, "demo_api" }
+        });
+
+        // ...
+
+```
+
+
+> [!WARNING] Use tunnel domain
+>
+> One pitfall with dev tunnels is, that they do not forward their domain name to the server but redirect traffic to `localhost`. This means, that even though you connect to your server using the dev tunnels URL (e.g. `https://vsr1d2gg-44359.euw.devtunnels.ms`), the request URL on your server will be `https://localhost:44359`. 
+>
+> This causes a variety of issues such as
+> - some OpenIddict WebProviders not working since the cookies provided by them in order to keep necessary state are lost since they are created for the tunnel domain and thus not served to `localhost`
+> - the OpenIddict client stack not being able to connect to your local server from a remote machine (e.g. Android emulator) since OpenId Connect Discovery fetches the `.well-known/openid-configuration` which will have all endpoints configured to `https://localhost:44359` instead of your dev tunnels URL.
```

Comment:
```suggestion
> This causes a variety of issues such as:
> - The request forgery protection feature not working correctly due to the antiforgery cookie created using the tunnel domain (and thus not served to `localhost`) being lost in the process.
>
> - The OpenIddict client stack not being able to connect to the server from a remote machine (e.g. Android emulator) due to the server metadata (`/.well-known/openid-configuration`) containing endpoints starting with `https://localhost:44359/` instead of the dev tunnel URL.
```